### PR TITLE
Added support for ephemeral sessions [SDK-1412]

### PR DESCRIPTION
**Pull request by Widcket**
### Changes

- Added a new boolean parameter called `ephemeralSession` to the `options` object that sets `prefersEphemeralWebBrowserSession` to `true` on iOS 13+. It has no effect on older versions of iOS, or on Android. **Usage of this method disables SSO.**

### References

Closes https://github.com/auth0/react-native-auth0/issues/294

### Testing

This change was tested manually by performing login and logout on iOS and Android.

- [X] This change adds unit test coverage
- [X] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] All existing and new tests complete without errors
- [X] All active GitHub checks have passed

### CODE REVIEW

File: src/webauth/agent.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
-import { NativeModules, Linking } from 'react-native';
+import {NativeModules, Linking, Platform} from 'react-native';
 
 export default class Agent {
-  show(url, closeOnLoad = false) {
+  show(url, ephemeralSession = false, closeOnLoad = false) {
```
Comment (lbalmaceda) [inline src/webauth/agent.js:4]: This agent class is public and I think a change like the one introduced here should be considered breaking. 

You are adding a parameter in the middle, those calling today with `show("URL", true) would have a different outcome. Please move it to the end and keep the defaults.

If instead, you consider this class as internal, let's document that and use a "configuration object" instead to make it future proof.

File: src/webauth/index.js [hunk 2]
```diff
@@ -41,15 +41,16 @@ export default class WebAuth {
...
+   * @param {String}  [parameters.nonce] Random string to prevent replay attacks of id_tokens.
+   * @param {String}  [parameters.audience] Identifier of Resource Server (RS) to be included as audience (aud claim) of the issued access token
+   * @param {String}  [parameters.scope] Scopes requested for the issued tokens. e.g. `openid profile`
+   * @param {String}  [parameters.connection] The name of the identity provider to use, e.g. "google-oauth2" or "facebook". When not set, it will display Auth0's Universal Login Page.
+   * @param {Number}  [parameters.max_age] The allowable elapsed time in seconds since the last time the user was authenticated (optional).
+   * @param {Object}  options Other configuration options.
+   * @param {Number}  [options.leeway] The amount of leeway, in seconds, to accommodate potential clock skew when validating an ID token's claims. Defaults to 60 seconds if not specified.
+   * @param {Boolean} [options.ephemeralSession] Disable SSO on iOS 13+. Has no effect on older versions of iOS.
```
Comment (lbalmaceda) [inline src/webauth/index.js:53]: I'd reword it to tell that only affects iOS and not Android. Use something like this.

```suggestion
   * @param {Boolean} [options.ephemeralSession] Disable Single-Sign-On (SSO). It only affects iOS with versions 13 and above.
```

File: src/webauth/agent.js (hunk 1 above)
Comment (Widcket) [reply src/webauth/agent.js:4]: I consider that class to be internal. How would a configuration object future-proof an internal class?

Comment (lbalmaceda) [reply src/webauth/agent.js:4]: My point is that if we keep adding parameters this signature won't look nice. It's OK because it's internal 👍 

A future-proof approach, that doesn't require changing the signature, would be:

```js
show(url, options = {})
const closeOnLoad = options.closeOnLoad || false
const ephemerealSession = options.ephemerealSession || false
```

File: src/webauth/agent.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
     if (!NativeModules.A0Auth0) {
       return Promise.reject(
         new Error(
-          'Missing NativeModule. React Native versions 0.60 and up perform auto-linking. Please see https://github.com/react-native-community/cli/blob/master/docs/autolinking.md.'
-        )
+          'Missing NativeModule. React Native versions 0.60 and up perform auto-linking. Please see https://github.com/react-native-community/cli/blob/master/docs/autolinking.md.',
```
Comment (lbalmaceda) [inline src/webauth/agent.js:8]: I think this extra comma was unintended 
```suggestion
          'Missing NativeModule. React Native versions 0.60 and up perform auto-linking. Please see https://github.com/react-native-community/cli/blob/master/docs/autolinking.md.'
```

File: src/webauth/index.js [hunk 4]
```diff
@@ -41,15 +41,16 @@ export default class WebAuth {
...
+   * @param {String}  [parameters.nonce] Random string to prevent replay attacks of id_tokens.
+   * @param {String}  [parameters.audience] Identifier of Resource Server (RS) to be included as audience (aud claim) of the issued access token
+   * @param {String}  [parameters.scope] Scopes requested for the issued tokens. e.g. `openid profile`
+   * @param {String}  [parameters.connection] The name of the identity provider to use, e.g. "google-oauth2" or "facebook". When not set, it will display Auth0's Universal Login Page.
+   * @param {Number}  [parameters.max_age] The allowable elapsed time in seconds since the last time the user was authenticated (optional).
+   * @param {Object}  options Other configuration options.
+   * @param {Number}  [options.leeway] The amount of leeway, in seconds, to accommodate potential clock skew when validating an ID token's claims. Defaults to 60 seconds if not specified.
+   * @param {Boolean} [options.ephemeralSession] Disable Single-Sign-On (SSO). It only affects iOS with versions 13 and above.
```
Comment (lbalmaceda) [inline src/webauth/index.js:53]: Thinking of the public API, and that "ephemeral session" is a concept that Apple introduced for their browser. Should we rename this to something more generic like "disableSSO" in case Android platform support it as well soon? We can keep a reference to ephemeral on the param doc/description

File: src/webauth/__tests__/agent.spec.js [hunk 5]
```diff
@@ -38,6 +38,22 @@ describe('Agent', () => {
         await agent.show(url);
         expect(A0Auth0.url).toEqual(url);
       });
+
+      it('should not use ephemeral session by default', async () => {
+        Platform.OS = 'ios';
```
[the remark this answers is not in the document]
Comment (Widcket) [reply src/webauth/__tests__/agent.spec.js:50]: Good catch. Added in https://github.com/auth0/react-native-auth0/pull/305/commits/405b4a2c975509ceafcdd39b7c469b87f4e0c9da.

File: src/webauth/agent.js (hunk 1 above)
Comment (Widcket) [reply src/webauth/agent.js:4]: Shouldn't we remove `closeOnLoad`, anyway? Seems like old leftover code that doesn't do anything now. The native iOS code receives it as a parameter but does nothing with it. Same with the Java code.

File: src/webauth/index.js (hunk 4 above)
Comment (Widcket) [reply src/webauth/index.js:53]: Given that "ephemeral session" is not a marketing term (unlike "Universal Links") and that the fact that it disables SSO is just an effect of a broader cause (not storing cookies), it's probably more accurate to leave it that way. It's generic enough IMO.

File: src/webauth/agent.js (hunk 1 above)
Comment (lbalmaceda) [reply src/webauth/agent.js:4]: iOS uses natively [here](https://github.com/auth0/react-native-auth0/blob/master/ios/A0Auth0.m#L216-L220). 

A side effect of the parameter is to know when to return a value. It's checked [here](https://github.com/auth0/react-native-auth0/blob/master/src/webauth/agent.js#L24-L28). From what I can tell, that's only in use for [authentication](https://github.com/auth0/react-native-auth0/blob/master/src/webauth/index.js#L72) and for [logout](https://github.com/auth0/react-native-auth0/blob/master/src/webauth/index.js#L131).

Comment (Widcket) [reply src/webauth/agent.js:4]: Good idea, but ATM it is just 3 parameters, which is fine IMO. If it keeps growing converting it to an options object is trivial.
